### PR TITLE
docs(examples): add section on working with elements

### PR DESCRIPTION
**Pull request by arjunattam**
With the goal of introducing the API to new comers, I've added a section to the getting started guide on elements, selectors and introduced some key API methods.

### CODE REVIEW

File: docs/examples/README.md [hunk 1]
```diff
@@ -43,6 +43,17 @@ It is also possible to open **browser developer tools** during execution, to ins
 * A [`BrowserContext`](../api.md#class-browsercontext) is an isolated incognito session within a browser instance. Browser contexts are fast to create and can be used to parallelize isolated test executions.
 * A [`Page`](../api.md#class-page) refers to a single tab within a browser context, which includes one or more [`Frame`](../api.md#class-frame) objects.
 
+### Working with elements
+
+* Types of selectors: Playwright can query elements on a web page through multiple selector engines like CSS, HTML attributes, XPath and text contents.
+  * See the [selectors guide](../selectors.md).
+* Actions on elements: Use methods like [`page.click`](../api.md#pageclickselector-options) or [`page.fill`](#pagefillselector-value-options) to execute actions on elements.
```
Comment (dgozman) [inline docs/examples/README.md:50]: `page.fill` link is broken.

File: docs/examples/README.md [hunk 2]
```diff
@@ -43,6 +43,17 @@ It is also possible to open **browser developer tools** during execution, to ins
...
+  * **Auto-wait** for elements: These actions auto-wait for the element to be ready.
+* Use the [`page.waitForSelector`](../api.md#pagewaitforselectorselector-options) method to explicitly wait for elements.
+* Assertions on elements: Use [`page.$`](../api.md#pageselector) to fetch the element from the page and [`page.$eval`](../api.md#pageevalselector-pagefunction-arg-1) to run a JS function in the execution context of the page.
+  * These can be used to assert the visibility or contents of the elements.
+  * The semantics for these methods are the same as the `$` in the DevTools console or in jQuery. They fetch the element from the page without waiting.
```
Comment (dgozman) [inline docs/examples/README.md:55]: Let's reiterate that most of the time you want `waitForSelector`.

File: docs/examples/README.md [hunk 3]
```diff
@@ -43,6 +43,17 @@ It is also possible to open **browser developer tools** during execution, to ins
 * A [`BrowserContext`](../api.md#class-browsercontext) is an isolated incognito session within a browser instance. Browser contexts are fast to create and can be used to parallelize isolated test executions.
 * A [`Page`](../api.md#class-page) refers to a single tab within a browser context, which includes one or more [`Frame`](../api.md#class-frame) objects.
 
+### Working with elements
```
Comment (dgozman) [inline docs/examples/README.md:46]: I think this section will benefit from small code snippet for each bullet.

**Comment by arjun27** [comment]
Thanks for the suggestions - made these changes. Please take another look

File: docs/examples/README.md [hunk 4]
```diff
@@ -25,24 +25,52 @@ const { chromium } = require('playwright');
...
+  ```js
+  // Wait for element to be ready, click it and wait for navigations (if any)
+  await page.click('text="Login"');
+  ```
+
+* Use the [`page.waitForSelector`](../api.md#pagewaitforselectorselector-options) method to explicitly wait for elements.
+
+  ```js
+  // Explicitly wait for element to be present in the DOM
+  await page.waitForSelector('data-test-id=foo');
```
Comment (dgozman) [inline docs/examples/README.md:62]: // Explicitly wait for the #search field to be present in the DOM
const search = await page.waitForSelector('#search');
// Retrieve the query
const query = await search.evaluate(element => element.value);

File: docs/examples/README.md [hunk 5]
```diff
@@ -25,24 +25,52 @@ const { chromium } = require('playwright');
...
+  ```
+
+* Assertions on elements: Use [`page.$`](../api.md#pageselector) to get the element from the page and [`page.$eval`](../api.md#pageevalselector-pagefunction-arg-1) to run a JS function in the execution context of the page.
+  * These can be used to assert element properties, like visibility or text contents.
+  * These methods behave similar to the `$` operator in DevTools console or jQuery. They fetch the element from the page without waiting. If required, use `page.waitForSelector` to explicitly wait here.
+
+  ```js
+  // Get value of the #search field
+  await page.$eval('#search', element => element.value);
```
Comment (dgozman) [inline docs/examples/README.md:71]: const query = await page.$eval('#search', element => element.value);

File: docs/examples/README.md [hunk 6]
```diff
@@ -25,24 +25,52 @@ const { chromium } = require('playwright');
...
+* Use the [`page.waitForSelector`](../api.md#pagewaitforselectorselector-options) method to explicitly wait for elements.
+
+  ```js
+  // Explicitly wait for element to be present in the DOM
+  await page.waitForSelector('data-test-id=foo');
+  ```
+
+* Assertions on elements: Use [`page.$`](../api.md#pageselector) to get the element from the page and [`page.$eval`](../api.md#pageevalselector-pagefunction-arg-1) to run a JS function in the execution context of the page.
+  * These can be used to assert element properties, like visibility or text contents.
+  * These methods behave similar to the `$` operator in DevTools console or jQuery. They fetch the element from the page without waiting. If required, use `page.waitForSelector` to explicitly wait here.
```
Comment (dgozman) [inline docs/examples/README.md:67]: If required, use `page.waitForSelector` and `evaluate` instead, as described above.

**Review by dgozman — APPROVED** [review]
Dropped some code suggestions, otherwise looks great.

**Pull request merged**